src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SignatureHelp/SignatureInformation.cs

```diff
@@ -1,14 +1,42 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using System;
+
 namespace Microsoft.PowerFx.Intellisense.SignatureHelp
 {
-    public class SignatureInformation
+    public class SignatureInformation : IEquatable<SignatureInformation>
     {
         public string Label { get; set; }
 
         public string Documentation { get; set; }
 
         public ParameterInformation[] Parameters { get; set; }
+
+        public bool Equals(SignatureInformation other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            // For now, label is enough to compare two signature information
+            if (Label != other.Label)
```

Comment:
I've been flip flopping on the logic here. In an ideal world, we should compare more attributes of the signature to decide whether they are equal or not and use those attributes to compute hash code as well. From my testing, I haven't found the case where two overloads map to the same signature/label but have different documentation/description. If that was the case, then we would need to consider documentation as well. Then there is active parameter and its label and its description. I don't think we need to implement such a complex comparison/hash code computation logic and Label should be fine. I leaned towards keeping it simple for now but let me know if you think i should make this more future proof. This simplicity was influenced by the testing i did and things i saw. But of course, someone with more knowledge about powerfx and its functions and their signatures can chime in here. @jas-valgotar wanted your attention here when you get a chance

Reply:
We need to also consider Documentation when comparing two signatures. 
I found examples for Boolean() function where two overloads have same signatures but different documentation
Look at the example below
<img width="737" alt="image" src="https://github.com/microsoft/Power-Fx/assets/25670945/f5352008-d83c-4c6d-a749-0633d6ec7a36">